src/dns/remote.rs

```diff
@@ -80,10 +81,20 @@ impl DnsResolver for RemoteResolver {
 
 impl RemoteResolver {
     /// Instantiates a new local override table
-    pub fn new(_opts: RemoteResolverOptions) -> RemoteResolver {
+    pub fn new(dns_opts: RemoteResolverOptions) -> RemoteResolver {
         // @todo: do something with the options
-        let config = ResolverConfig::default();
-        let opts = ResolverOpts::default();
+        let mut config = ResolverConfig::default();
+        let mut opts = ResolverOpts::default();
+
+        for nameserver in dns_opts.nameservers.iter() {
+            if let Ok(ip) = IpAddr::from_str(nameserver.as_str()) {
```

Comment:
I guess, you don't need `as_str` here, because it is already a reference to a string. `&String` also counts as `&str`. But feel free to ignore.

Reply:
I'd use `IpAddr::from_str(&nameserver)` and only fall back to `as_str()` if the compiler complains.